### PR TITLE
feat(fulgur-wpt): WPT reftest runner Phase 1 finish (2foo.7-9)

### PR DESCRIPTION
**Pull request by mitsuru**
## 概要

PR #134 (Phase 1 core) の続編。WPT reftest runner の Phase 1 を完結させます。具体的には wptreport.json 出力、css-page 実走行による expectations の初期 seed、CI (PR job + nightly) 統合の 3 セットです。

**Epic:** fulgur-2foo
**Beads issues closed:** fulgur-2foo.7, .8, .9
**Next PR:** Phase 2 tracker (fulgur-2foo.10 css-break) から着手予定

## 成果物サマリ

| 追加 | 役割 |
|---|---|
| `src/report.rs` | wptreport.json エミッター (5 unit tests). `push()`/`push_error()` で FAIL vs ERROR を分離 |
| `examples/seed.rs` | WPT サブディレクトリを走査し expectations/<subdir>.txt を自動生成する one-shot bin |
| `expectations/css-page.txt` | **66 PASS / 111 FAIL / 14 SKIP** (total 191) の初期 seed |
| `tests/wpt_css_page.rs` | Phase 1 実走 integration test。expectations と突き合わせて regression のみ hard fail |
| `.github/workflows/ci.yml` | `wpt-css-page` PR job (poppler install + fetch + 実テスト) |
| `.github/workflows/wpt-nightly.yml` | 02:00 UTC cron で全量実走 → artifact + `gh issue create` on failure |
| `scripts/wpt/pinned_sha.txt` | `DEADBEEF...` → 実 WPT SHA `97ea26e2` |
| `crates/fulgur-wpt/README.md` | Expectations 運用フロー (seed → 実測で判定 → PASS 昇格 / SKIP 一時除外) |

## 設計上のポイント

- **`ERROR` status を FAIL と分離**: `push_error()` を新設。wpt.fyi が ERROR と FAIL を別バケットで扱うため、harness クラッシュとテスト失敗を意味論的に区別
- **`catch_unwind(AssertUnwindSafe(...))`**: blitz-dom 内の `unwrap()` パニックで seed や integration test 全体が止まらないように、各テスト呼び出しをパニックセーフにラップ
- **`workspace_root()` helper**: integration test の CWD は `CARGO_MANIFEST_DIR` なので `env!("CARGO_MANIFEST_DIR")` から workspace root を解決、`target/wpt` と expectations へのパスを正しく引く
- **`pub(crate)` で `TestResult` を封印**: 外部から `status: "BOGUS"` と直接構築できないよう可視性を絞り、`push()` / `push_error()` 経由のみを強制
- **seed の自動 header**: `# Summary: P PASS, Q FAIL, R SKIP (total N)` を先頭に記録、運用時の at-a-glance に便利

## テスト / ゲート

- `cargo test -p fulgur-wpt`: **56 tests** (49 lib + 3 harness_smoke + 1 render_multi_page + 1 diff_pages + 1 wpt_css_page + 1 wpt_smoke) 全 pass
- `cargo test -p fulgur --lib`: 549 pass (baseline 回帰なし)
- `cargo test -p fulgur-vrt --lib`: 12 pass
- `cargo fmt --check` / `cargo clippy -p fulgur-wpt --all-targets -- -D warnings`: clean
- markdownlint (plan + 2 READMEs): 0 errors
- `wpt_css_page` 実走: 177 ok + 14 skipped = 191、**0 regressions / 0 promotions / 0 unknown**

## PASS 数が意外に多い (66/191) 件

`basic-pagination-001` 等、fulgur が CSS `break-after: page` を未配線のためテスト側も ref 側も同じ単ページレンダリングになって false pass しているケースが複数ある想定です。Phase 1 のスコープは「seed を捕捉して運用ベースラインを確立する」ことなので、現状のまま commit しています。fulgur 本体の `page-break-after` 配線は別 issue (fulgur-lje5) で追跡済み。

## フォローアップ TODO (本 PR では入れない)

- Phase 2 (2foo.10 css-break ~968 reftest) 着手
- `WptReport` の実際の書き出し call site (現状は future consumer 向け API のみ)
- examples/seed.rs の clap 化 (flag 追加時の壊れやすさ対策)
- label `wpt-nightly-regression` の手動作成 (nightly が実際に回った後で対応)

## Test Plan

- [x] `cargo test -p fulgur-wpt` 全 pass
- [x] baseline 回帰なし (fulgur 549, fulgur-vrt 12)
- [x] `cargo fmt --check` + `cargo clippy -p fulgur-wpt --all-targets -- -D warnings`
- [x] markdownlint on plan + 2 READMEs
- [x] `python3 -c "import yaml; yaml.safe_load(...)"` on both workflow files
- [x] `wpt_css_page` local dry-run: 0 regressions
- [ ] (merge 前) CI で `wpt-css-page` job が green を確認

## 参考資料

- 実装プラン: `docs/plans/2026-04-21-wpt-runner-phase1-finish.md`
- 元設計: `docs/plans/2026-04-21-wpt-reftest-runner-design.md`

## Summary by CodeRabbit

* **New Features**
  * Web Platform Tests for CSS paged-media with JSON test report output
  * Nightly runs that collect logs/artifacts and file regression issues automatically

* **Tests**
  * Integration test validating css-page rendering against declared expectations
  * CI job added to run CSS page tests on PRs

* **Documentation**
  * Added expectations management and seed-generation workflow

* **Chores**
  * Pinned WPT snapshot to a concrete commit

### CODE REVIEW

File: .github/workflows/wpt-nightly.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        if: steps.run.outcome == 'failure'
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh issue create \
+            --title "WPT nightly regression $(date -u +%Y-%m-%d)" \
+            --body "Nightly WPT run detected regressions on css-page. See run: $GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID
+
+          Download the \`wpt-logs\` artifact from the run for the full log and diff images." \
+            --label "wpt-nightly-regression" || true
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply .github/workflows/wpt-nightly.yml]: 053c07b で修正しました。提案いただいた conditional パターンを採用し、`gh label list --search` でラベル存在をチェックしてから、あれば `--label` 付き、なければ無しで `gh issue create` を実行する形に変更しました。さらに step 全体に `set -euo pipefail` を追加して `gh issue create` 自体の失敗は silent に飲まないようにしています。

File: crates/fulgur-wpt/examples/seed.rs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    let mut tests: Vec<PathBuf> = std::fs::read_dir(&dir)?
+        .filter_map(|e| e.ok())
+        .map(|e| e.path())
+        .filter(|p| {
+            let name = p.file_name().and_then(|s| s.to_str()).unwrap_or("");
+            name.ends_with(".html")
+                && !name.ends_with("-ref.html")
+                && !name.ends_with("-notref.html")
+        })
+        .collect();
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur-wpt/examples/seed.rs]: 053c07b で修正しました。`crates/fulgur-wpt/src/reftest.rs` に共有ヘルパー `pub fn collect_reftest_files(root: &Path) -> io::Result<Vec<PathBuf>>` を新設し、recursive walk + stem ベースのフィルタ (後述 thread 3 の指摘対応) + `reference` / `resources` / `support` サブディレクトリの除外を一箇所に集約しました。seed.rs と wpt_css_page.rs の両方がこれを使う形にリファクタしています。結果として seed entries が 191 → 257 に増加 (主に `tentative/` 配下のテストを拾うようになった分) し、`-ref.tentative.html` の leak も消えました。

File: crates/fulgur-wpt/expectations/css-page.txt [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+FAIL  css/css-page/page-margin-003-print.html  # page 1 diff: 66780/891662 pixels exceed tol (max_ch=255)
+FAIL  css/css-page/page-margin-004-print.html  # page 1 diff: 67614/90000 pixels exceed tol (max_ch=255)
+FAIL  css/css-page/page-margin-005-print.html  # page 1 diff: 51597/180000 pixels exceed tol (max_ch=255)
+FAIL  css/css-page/page-margin-006-print.html  # page 1 diff: 39226/331776 pixels exceed tol (max_ch=255)
+FAIL  css/css-page/page-margin-007-print.html  # page 1 diff: 39964/80000 pixels exceed tol (max_ch=255)
+FAIL  css/css-page/page-margin-auto-and-non-zero-print.html  # page 1 diff: 163324/891662 pixels exceed tol (max_ch=255)
+SKIP  css/css-page/page-margin-auto-negative-print-ref.tentative.html  # NoMatch
+FAIL  css/css-page/page-margin-auto-negative-print.tentative.html  # page 1 diff: 56325/90000 pixels exceed tol (max_ch=255)
+FAIL  css/css-page/page-margin-auto-print.html  # page 1 diff: 112431/891662 pixels exceed tol (max_ch=255)
+SKIP  css/css-page/page-margin-negative-print-ref.tentative.html  # NoMatch
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur-wpt/expectations/css-page.txt]: 053c07b で修正しました。提案いただいた stem ベースのフィルタを採用: `name.strip_suffix(".html")` → `split('.').next()` で `.tentative` 等のフラグを剥がした後の base に対して `ends_with("-ref")` / `ends_with("-notref")` を判定しています。これを `collect_reftest_files()` (reftest.rs に新設) に集約し、`collect_reftest_files_recurses_and_filters` ユニットテストで `-ref.tentative.html` を含む edge case を pin しました。expectations/css-page.txt を再生成済み (`page-margin-auto-negative-print-ref.tentative.html` 等の 2 件は消えています)。

File: crates/fulgur-wpt/tests/wpt_css_page.rs [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        .filter_map(|e| e.ok())
+        .map(|e| e.path())
+        .filter(|p| {
+            let name = p.file_name().and_then(|s| s.to_str()).unwrap_or("");
+            name.ends_with(".html")
+                && !name.ends_with("-ref.html")
+                && !name.ends_with("-notref.html")
+        })
+        .collect();
+    tests.sort();
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply crates/fulgur-wpt/tests/wpt_css_page.rs]: 053c07b で修正しました。thread 2 と同じ `collect_reftest_files()` 共有ヘルパーを integration test 側でも使うようにリファクタ。recursive 化により tests 数が 191 → 257 に増え、nested 配下の tentative/margin-boxes/cssom 等も走るようになりました。local dry-run で 219 ok + 38 skipped = 257、0 regressions を確認済みです。

File: docs/plans/2026-04-21-wpt-runner-phase1-finish.md [hunk 5]
```diff
@@ -0,0 +1,852 @@
...
+---
+
+## Task 2: pinned_sha.txt を実 WPT SHA に差し替え
+
+**Beads:** fulgur-2foo.8 (前提作業)
+
+**Files:**
+
+- Modify: `scripts/wpt/pinned_sha.txt`
+- Modify: `scripts/wpt/README.md` (plcaholder 文言を削除)
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply docs/plans/2026-04-21-wpt-runner-phase1-finish.md]: 053c07b で修正しました。`plcaholder` → `placeholder`。

File: .github/workflows/wpt-nightly.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mitsuru) [reply .github/workflows/wpt-nightly.yml]: 053c07b ですでに提案いただいた conditional パターンを採用しています (`gh label list --search ... | grep -q; then --label else 無し fi`)。`|| true` も除去済み。step 全体に `set -euo pipefail` を入れて `gh issue create` 自体の失敗は surface させる形にしています。再指摘は旧 diff 位置の stale cache 起因と思われるので、現行 `.github/workflows/wpt-nightly.yml:43-60` をご確認ください。

File: docs/plans/2026-04-21-wpt-runner-phase1-finish.md [hunk 6]
```diff
@@ -0,0 +1,852 @@
+# WPT Runner Phase 1 Finish Implementation Plan
+
+> **For Claude:** REQUIRED SUB-SKILL: Use superpowers:executing-plans to implement this plan task-by-task.
+
+**Goal:** WPT CSS paged media reftest runner の Phase 1 を完成させる — `wptreport.json` 出力、css-page 全 reftest を実走行した expectations の初期 seed、CI ワークフロー統合を入れる。
+
+**Architecture:** Phase 1 core (PR #134) で完成した `crates/fulgur-wpt/` に `report.rs` を追加、`scripts/wpt/pinned_sha.txt` を実 SHA に差し替え、`scripts/wpt/seed.sh` で css-page を一括実行して `expectations/css-page.txt` を生成、`.github/workflows/ci.yml` に PR job、新規 `wpt-nightly.yml` に cron workflow を追加する。
```
[the remark this answers is not in the document]
Comment (mitsuru) [reply docs/plans/2026-04-21-wpt-runner-phase1-finish.md:7]: Dismiss します。指摘の lines 7, 543, 787 の "github" は、いずれも `.github/workflows/ci.yml` や `.github/workflows/wpt-nightly.yml` というファイルシステム上のパスを指しており、GitHub Actions のディレクトリ名は **常に小文字** の `.github/` です (GitHub docs および repo 内の既存参照と一貫)。line 787 の散文は `GitHub Actions は YAML 構文が厳格` と既に大文字で書かれているので、誤検知と判断します。
